From 9631ca0bc102f351fa07f2d0a88f963f7d4e7441 Mon Sep 17 00:00:00 2001
From: Oliver Fuhrer <oli@ginko.ch>
Date: Fri, 8 May 2020 12:38:03 -0700
Subject: [PATCH] add sat_vapor_pres to FMS

---
 libFMS/Makefile.am | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libFMS/Makefile.am b/libFMS/Makefile.am
index 68d237c..6aefef4 100644
--- a/libFMS/Makefile.am
+++ b/libFMS/Makefile.am
@@ -87,6 +87,8 @@ libFMS_la_LIBADD += ${top_builddir}/exchange/libxgrid.la
 libFMS_la_LIBADD += ${top_builddir}/topography/libtopography.la
 libFMS_la_LIBADD += ${top_builddir}/tracer_manager/libtracer_manager.la
 libFMS_la_LIBADD += ${top_builddir}/station_data/libstation_data.la
+libFMS_la_LIBADD += ${top_builddir}/sat_vapor_pres/libsat_vapor_pres.la
+libFMS_la_LIBADD += ${top_builddir}/sat_vapor_pres/libsat_vapor_pres_k.la
 
 # At least one source file must be included to please Automake.
 libFMS_la_SOURCES = ${top_builddir}/include/file_version.h
-- 
2.24.0

